dsa.go

```diff
@@ -0,0 +1,331 @@
+//go:build !cmd_go_bootstrap
+
+package openssl
+
+// #include "goopenssl.h"
+import "C"
+import (
+	"runtime"
+	"unsafe"
+)
+
+var (
+	paramPbits = C.CString("pbits")
+	paramQbits = C.CString("qbits")
+	paramP     = C.CString("p")
+	paramQ     = C.CString("q")
+	paramG     = C.CString("g")
```

Comment:
Should these names differentiate that they're for DSA? E.g., we also have `paramRSA_Q` which is `rsa-factor2`. If a future PR accidentally uses `paramQ` in an RSA context, it would be easier to spot (or wouldn't occur) if this were called `paramDSA_Q`.

On the other hand, since it's literally `"q"`, maybe it should be intentionally shared if need be, and be moved to a different file than `dsa.go` if that happens?

Reply:
This raises a valid point. I'll switch the naming pattern to use the names given in the OpenSSL source code (e.g. `OSSL_PKEY_PARAM_FFC_PBITS` instead of `pbits`. Node that DSA parameters are shared with other FFC algorithms, but we don't implement more of those yet, so we can keep this variable here for the moment.

I'll update the parameter names for other algorithms (ECDSA, HMAC and RSA) in a separate PR.